### DISCUSSION TITLE
Allow `smooth` scroll with dynamic size

### DISCUSSION THREAD

**Discussion by egorprnn** [TanStack/virtual#495] [General] [answered]
As part of the changes: https://github.com/TanStack/virtual/pull/469, smooth scrolling has been disabled when elements are dynamically sized. It didn't really work if the element was not in view, but for other elements this restriction is meaningless.
Maybe it's worth checking if the element is in `virtualizer.range` and if it is, then allow smooth scrolling?

**Comment by piecyk** [comment, marked as answer]
Overall there are few issues when we scroll to index, basic an offset in list where we didn't be before.

* fist is the issue that offset of the index will change as scrolling, new items will be rendered and real size will change the current measurements done initialy via estimateSize
* second is when real size is different from estimateSize of items before scroll offset, we need to adjust scroll position with delta to keep the items in place. In chrome it will break the current smooth scroll.

Because of this decided to disable the behavior smooth in dynamic size, change it in latest beta as better just warn users that native smooth scroll is not fully supported than disable it. 

Also with latest beta, you can write your own smooth scroll like we had in v2 https://codesandbox.io/s/quizzical-mcclintock-r98ru9?file=/src/main.tsx

Current testing the idea to disable the size adjustment till scroll happen.
